src/sgame/sg_bot_nav.cpp

```diff
@@ -830,17 +830,37 @@ Global Bot Navigation
 =========================
 */
 
+static constexpr int WALLCLIMB_MAX_TIME = 3000;
+
 static Cvar::Cvar<int> g_bot_upwardNavconMinHeight("g_bot_upwardNavconMinHeight", "minimal height difference for bots to use special upward movement.", Cvar::NONE, 100);
 static Cvar::Cvar<int> g_bot_upwardLeapAngleCorr("g_bot_upwardLeapAngleCorr", "is added to the angle for mantis and dragoon when leaping upward (in degrees).", Cvar::NONE, 20);
 
+static void BotClimbToGoal( gentity_t *self )
```

Comment:
I would just check for `BG_ClassHasAbility( pcl, SCA_WALLCLIMBER )` and return. This way, the code would be more generic, and this could be moved out of the switch. Notably, it would enable adv grangering when we'll have building bots.

Reply:
Not really. That would work for dretch and advanced granger. The mantis, however, will have to decide if a leap or a wall climb is the way to go. Right now, it goes for the wall climb if over a navcon, and for the leap if attacking. That is not optimal.